src/logic/map_objects/tribes/ship.cc

```diff
@@ -311,6 +311,22 @@ void Ship::recalc_expedition_swimmable(const EditorGameBase& egbase) {
 	}
 }
 
+bool Ship::suited_as_invasion_portspace(const Coords& coords) const {
+	// We can invade any port space, regardless of owner and immovables.
+	// But we ignore port spaces where we already have an own port.
+
+	const EditorGameBase& egbase = owner().egbase();
+	const Map& map = egbase.map();
+	if (!map.is_port_space(coords)) {
+		return false;
+	}
+
+	const Field& field = map[coords];
+	return (field.get_immovable() == nullptr || field.get_immovable()->get_owner() != get_owner() ||
+	        egbase.descriptions().building_index(field.get_immovable()->descr().name()) !=
+	           owner().tribe().port());
```

Comment:
E.g. the map "The Nile" has several portspaces packed next to each other. If it's not too hard, this should also return false if there's an own portspace on some nearby node. (within radius of 5? – yes, that's `kPortSpaceGeneralAreaRadius` for soldiers already)

(It's rather subtle, but it still may make sense to bring soldiers as reinforcement if a connected economy has warehouses. Maybe less so if it has ports too, but even those may be too far away. So no need to use the same check here as for soldiers ending the invasion.)

Reply:
Thanks.